File: Domain/Entities/Drink.swift
//
//  Drink.swift
//  Domain
//
//  Created by Balázs Kilvády on 2/17/20.
//  Copyright © 2020 kil-dev. All rights reserved.
//

import Foundation

public struct Drink: Codable {
    typealias ID = Int64

    let id: ID
    let name: String
    let price: Double
}

File: Domain/Sources/CppCart/Drink.cpp
//
//  Drink.cpp
//  DCPizza
//
//  Created by Balázs Kilvády on 02/09/21.
//  Copyright © 2021. kil-dev. All rights reserved.
//

#include "Drink.hpp"
#include <iostream>

namespace cpplib {

Drink::Drink(ID id,
             const string& name,
             double price)
: id(id), name(name), price(price)
{
    std::cout << "Drink created with: ("
    << id << ", "
    << name << ", "
    << price << ")\n";
}

Drink::~Drink()
{
    std::cout << "Drink destroyed with: ("
    << id << ", "
    << name << ", "
    << price << ")\n";
}

}
